pyproject.toml

```diff
@@ -85,55 +85,50 @@ jinja2-time = { version = "^0.2.0", optional = true }
-coverage = { extras = ["toml"], version = "^5.5" }
-pre-commit = "^2.14.0"
-pyment = "^0.3.3"
-tox = "^3.24.3"
-hypothesis = "^6.43.1"
-typing-extensions = ">=3.7.4"
-darglint = "^1.8.1"
-
-# pytest dev dependencies
-pytest-randomly = "^3.10.1"
-pytest-mock = "^3.6.1"
-pytest-clarity = "^1.0.1"
-
+# Optional development dependencies
+coverage = { extras = ["toml"], version = "^5.5", optional = true }
+mypy = { version = "^1.0.0", optional = true }
+pre-commit = { version = "^2.14.0", optional = true }
+pyment = { version = "^0.3.3", optional = true }
+tox = { version = "^3.24.3", optional = true }
+hypothesis = { version = "^6.43.1", optional = true }
+typing-extensions = { version = ">=3.7.4", optional = true }
+darglint = { version = "^1.8.1", optional = true }
+# pytest
+pytest = { version = "^6.2.4", optional = true }
+pytest-randomly = { version = "^3.10.1", optional = true }
+pytest-mock = { version = "^3.6.1", optional = true }
+pytest-clarity = { version = "^1.0.1", optional = true }
 # mkdocs including plugins
-mkdocs="^1.2.3"
-mkdocs-material="^8.1.7"
-mkdocs-awesome-pages-plugin="^2.6.1"
-mkdocstrings="^0.17.0"
-mike="^1.1.2"
-
+mkdocs = { version = "^1.2.3", optional = true }
+mkdocs-material = { version = "^8.1.7", optional = true }
+mkdocs-awesome-pages-plugin = { version = "^2.6.1", optional = true }
+mkdocstrings = { version = "^0.17.0", optional = true }
+mike = { version = "^1.1.2", optional = true }
 # mypy type stubs
-types-certifi = "^2021.10.8.0"
-types-croniter = "^1.0.2"
-types-futures = "^3.3.1"
-types-Markdown = "^3.3.6"
-types-Pillow = "^9.2.1"
-types-protobuf = "^3.18.0"
-types-PyMySQL = "^1.0.4"
-types-python-dateutil = "^2.8.2"
-types-python-slugify = "^5.0.2"
-types-PyYAML = "^6.0.0"
-types-redis = "^4.1.19"
-types-requests = "^2.27.11"
-types-setuptools = "^57.4.2"
-types-six = "^1.16.2"
-types-termcolor = "^1.1.2"
-types-psutil = "^5.8.13"
-
+types-certifi = { version = "^2021.10.8.0", optional = true }
+types-croniter = { version = "^1.0.2", optional = true }
+types-futures = { version = "^3.3.1", optional = true }
+types-Markdown = { version = "^3.3.6", optional = true }
+types-Pillow = { version = "^9.2.1", optional = true }
+types-protobuf = { version = "^3.18.0", optional = true }
+types-PyMySQL = { version = "^1.0.4", optional = true }
+types-python-dateutil = { version = "^2.8.2", optional = true }
+types-python-slugify = { version = "^5.0.2", optional = true }
+types-PyYAML = { version = "^6.0.0", optional = true }
+types-redis = { version = "^4.1.19", optional = true }
+types-requests = { version = "^2.27.11", optional = true }
+types-setuptools = { version = "^57.4.2", optional = true }
+types-six = { version = "^1.16.2", optional = true }
+types-termcolor = { version = "^1.1.2", optional = true }
+types-psutil = { version = "^5.8.13", optional = true }
 # test framework dependencies
-docker-compose = "^1.28"
+docker-compose = { version = "^1.28", optional = true }
+
+[tool.poetry.extras]
```

Comment:
why is this still referring to Poetry?

Reply:
This is for the publishing to pypi which still uses poetry. However this poetry value is a synonym that also works for pip (See https://peps.python.org/pep-0621/)